### ISSUE TITLE
When disabling editor it is re-enabled when clicked

### ISSUE THREAD

**Issue by fran-worley** [froala/wysiwyg-editor#4110]
##### Expected behavior.
When editing is turned off, it should remain disabled even when the page is clicked on.

##### Actual behavior.
On page click (anywhere on the page), the editor is re-enabled.

##### Steps to reproduce the problem.
Technically we are using ember-froala-editor but I understand from our technical team that the issue is more likely to be with Froala itself (hence I have raised it in this repo). 

When rendering the editor, we set editor event `editor.edit.off()` see https://github.com/froala/ember-froala-editor/blob/b750125afc3f7b39ce9bb1372011694ae69a8e6d/addon/components/froala-editor.js#L225

Then once the page has rendered, click anywhere on the page. The editor should remain 'off' but turns itself back on.

##### Editor version.
3.2.3

##### OS.
10.15.7

##### Browser.
Chrome 87.0.4280.88

##### Recording.
![2020-12-09_15-39-05 (1)](https://user-images.githubusercontent.com/2316837/101651343-dddeab00-3a34-11eb-9ca8-cb648c67a18c.gif)

**Comment by awegmueller** [comment]
Same issue here with v3.2.3. I tracked down the issue to this change in froala_editor_pkgd.js. On line 504017 this block has been added: 
```
// If there are no cells selected, enable editing


      if (!selectedCells().length) {
        // Allow text selection.
        editor.$el.removeClass('fr-no-selection');
        editor.edit.on();
      }
```

This re-enables the editor on each click. I made a patch by changing Line 50420 to:

`if (!editor.edit.isDisabled() && !selectedCells().length) {`

This works for me, but I'm not sure if it destroys a regression test somewhere else. The issue can be reproduced on the Froala editor homepage. Execute this in your browser console:

`$('#MAIN-FROALA-EDITOR')[0]['data-froala.editor'].edit.off()`

Click on the editor --> editor will be enabled (which is wrong).

**Comment by Krisell** [comment]
Perhaps this issue was introduced when fixing this: https://github.com/froala/wysiwyg-editor/issues/3615

**Comment by fran-worley** [comment]
This was fixed in v3.2.5

**Comment by Krisell** [comment]
Please note that this, rather serious, issue is present in 3.2.5: https://github.com/froala/wysiwyg-editor/issues/3615
